app/controllers/concerns/accounts/omniauth_login.rb

```diff
@@ -47,8 +47,12 @@ module Accounts::OmniauthLogin
   end
 
   def omniauth_login
-    # Set back url to page the omniauth login link was clicked on
-    params[:back_url] = request.env['omniauth.origin']
+    # Remmember the back_url to redirect to after login
+    # only if we're in a direct login phase, so the user ends up
+    # in the original requested URL after logging in
+    if omniauth_direct_login?
+      params[:back_url] = request.env['omniauth.origin']
+    end
```

Comment:
FYI @machisuji @ulferts  This change is necessary to prevent users from being redirected back to the back_url now that redirect_after_login uses the back_url presence. We need it for the sharing as an invited user will have this flow:

1. Click /account/activate?token=foo&back_url=/work_packages/bla
2. Do the registration form or whatever flow, save the back_url in the session of `AccountController#activate_invited`
3. Come back from authentication stages, expect to be redirected to work packages page

5. Come back